### ISSUE TITLE
Attributes on struct rest patterns are ignored

### ISSUE THREAD

**Issue by Skepfyr** [rust-lang/rust#136108]
Attributes on struct field rest patterns are entirely ignored by the compiler, in fact they're dropped by the parser, so they can't generate any errors (beyond parse errors) but are accepted by the parser.

Specifically, this bit of the AST doesn't store the attributes:
https://github.com/rust-lang/rust/blob/01a26c026ddd9f0da0307468b07b003b8759fc4a/compiler/rustc_ast/src/ast.rs#L845-L854

The parser drops them on the floor [here](https://github.com/rust-lang/rust/blob/01a26c026ddd9f0da0307468b07b003b8759fc4a/compiler/rustc_parse/src/parser/pat.rs#L1535).

This means that the following code doesn't produce any errors:
```rust
let Foo { #[bad] .. } = Foo {};
```
And entertainingly the attribute is deleted by rustfmt...I assume that's because it's not represented in the AST.

Now, as far as I know there aren't any attributes that would be reasonable to use in this position, which is probably why the bug still exists.

**Comment by ehuss** [comment]
> Now, as far as I know there aren't any attributes that would be reasonable to use in this position, which is probably why the bug still exists.

I would expect `#[cfg]` to work there, but it is also ignored.

**Comment by petrochenkov** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/81282.
I think my answer in https://github.com/rust-lang/rust/issues/81282#issuecomment-765662665 is still relevant.

**Comment by ehuss** [comment]
Ah, I knew there was an issue somewhere, but I was searching for the word "rest" and wasn't finding that. Thanks!
